FileIO/VtkIO/PVDFile.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <limits>
+#include <logog/include/logog.hpp>
+
+#include "PVDFile.h"
+
+namespace FileIO
+{
+
+PVDFile::PVDFile(const std::string &fn)
+    : _fh{fn.c_str()}
+{
+    if (!_fh) {
+        ERR("could not open file `%s'", fn.c_str());
+        std::abort();
+    }
+
+    _fh << std::setprecision(std::numeric_limits<double>::digits10);
+
+    _fh << "<?xml version=\"1.0\"?>\n"
+           "<VTKFile type=\"Collection\" version=\"0.1\" byte_order=\"LittleEndian\""
+           " compressor=\"vtkZLibDataCompressor\">\n"
+           "  <Collection>\n";
+}
+
+void PVDFile::addVTUFile(const std::string &fn, double timestep)
+{
+    _fh << "    <DataSet timestep=\"" << timestep << "\" group=\"\" part=\"0\" file=\"" << fn << "\"/>\n";
+}
+
+PVDFile::~PVDFile()
+{
+    _fh << "  </Collection>\n</VTKFile>\n";
```

Comment:
I don't see any problem.

> Destructor should be used only for finalizing resources.

It only finalizes the XML data.

> Moreover, it's not an intuitive interface. A user of this class has to delete an object to get a complete output.

I think, there's no problem. Actually: That's rather intuitive. Since objects are destructed automatically, this implementation asserts that you cannot forget to call some `close()` routine.